### PR TITLE
Schemaless tap

### PR DESCRIPTION
**Pull request by JingLin0**
# Description of change
(write a short description or paste a link to JIRA)

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_intercom/stream.py [hunk 1]
```diff
@@ -16,25 +16,18 @@ def __init__(self, config: Dict):
...
+            try:
+                data = self.intercom.get_records(tap_stream_id)
+                for record, replication_value in data:
 
                         if replication_value and (
                             start_date >= replication_value or end_date <= replication_value
                         ):
                             continue
 
-                        record = transformer.transform(d, self.schema, self.mdata)
```
Comment (przywartya) [inline tap_intercom/stream.py:50]: Why not use `transform`?

File: tap_intercom/stream.py [hunk 2]
```diff
@@ -16,25 +16,18 @@ def __init__(self, config: Dict):
...
-                        singer.write_record(self.tap_stream_id, record)
-                        counter.increment(1)
```
Comment (przywartya) [inline tap_intercom/stream.py:52]: You are not using `counter.increment` now. Is this OK? Maybe remove also `with singer.metrics.record_counter(tap_stream_id) as counter:` if we don't need counter

File: tap_intercom/stream.py [hunk 3]
```diff
@@ -16,34 +17,49 @@ def __init__(self, config: Dict):
...
-
-    def __get_start_end(self, state: dict):
+                    if replication_value and (
+                        start_date >= replication_value or end_date <= replication_value
+                    ):
+                        continue
+
+
+                    singer.write_record(tap_stream_id, record)
+                    counter.increment(1)
```
Comment (przywartya) [inline tap_intercom/stream.py:36]: OK!

File: tap_intercom/stream.py [hunk 4]
```diff
@@ -16,34 +17,49 @@ def __init__(self, config: Dict):
...
+                        bookmark_key=BOOKMARK_KEY,
+                        tap_stream_id=tap_stream_id,
+                    )
+                    prev_bookmark = new_bookmark
+
+            finally:
+                self.__advance_bookmark(
+                    state=state,
+                    bookmark=prev_bookmark,
+                    bookmark_key=BOOKMARK_KEY,
```
Comment (przywartya) [inline tap_intercom/stream.py:58]: If `bookmark_key` is a global value `BOOKMARK_KEY` then we don't need to pass it as argument, we can just use it in `__advance_bookmark` function :)

File: tap_intercom/stream.py (hunk 1 above)
Comment (JingLin0) [reply tap_intercom/stream.py:50]: https://www.notion.so/dreamdata/Law-of-the-Taps-abac8f608b61457dbe47b648a6216af3#33f24a6ee5f94b4382b97bfa12f6da11

**Pull request merged**